lib/modules.nix

```diff
@@ -669,22 +667,25 @@ rec {
   };
 
   doRename = { from, to, visible, warn, use }:
+    { config, options, ... }:
     let
-      toOf = attrByPath to
-        (abort "Renaming error: option `${showOption to}' does not exist.");
+      fromOpt = getAttrFromPath from options;
+      toOpt = getAttrFromPath from options;
+      toOf = attrByPath to (abort "Renaming error: option `${showOption to}' does not exist.");
     in
-      { config, options, ... }:
-      { options = setAttrByPath from (mkOption {
-          inherit visible;
-          description = "Alias of <option>${showOption to}</option>.";
-          apply = x: use (toOf config);
-        });
-        config = {
-          warnings =
-            let opt = getAttrFromPath from options; in
-            optional (warn && opt.isDefined)
-              "The option `${showOption from}' defined in ${showFiles opt.files} has been renamed to `${showOption to}'.";
-        } // setAttrByPath to (mkAliasDefinitions (getAttrFromPath from options));
-      };
+    {
+      options = setAttrByPath from (mkOption {
+        inherit visible;
+        description = "Alias of <option>${showOption to}</option>.";
+        apply = x: use (toOf config);
+      });
+      config = mkMerge [
+        {
+          warnings = optional (warn && toOpt.isDefined)
```

Comment:
This should be `fromOpt` and not `toOpt`. We want to warn uses of the renamed option, not uses of the new option.